cns/restserver/ipam.go

```diff
@@ -176,6 +171,9 @@ func (service *HTTPRestService) requestIPConfigsHandler(w http.ResponseWriter, r
 		// Wrap the default datapath handlers with the middleware
 		wrappedHandler := service.IPConfigsHandlerMiddleware.IPConfigsRequestHandlerWrapper(service.requestIPConfigHandlerHelper, service.releaseIPConfigHandlerHelper)
 		ipConfigsResp, err = wrappedHandler(r.Context(), ipconfigsRequest)
+		if ipconfigsRequest.AddInterfacesDataToResponse {
```

Comment:
Currently we are passing in ipconfigsRequest by value, so whatever changes we made to ipconfigsRequest within the wrapped handler will not stick when we exited to here. You can change the `IPConfigsHandlerFunc` function signature to taking in `IPConfigsRequest` as a pointer instead.

Reply:
good catch, I've modified it for k8s middleware too